src/main/java/org/openrewrite/java/migrate/javax/AddDefaultConstructorToEntityClass.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import lombok.EqualsAndHashCode;
+import org.openrewrite.*;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+
+import java.util.Comparator;
+
+@EqualsAndHashCode(callSuper = false)
+public class AddDefaultConstructorToEntityClass extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "`@Entity` objects with constructors must also have a default constructor";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When a Java Persistence API (JPA) entity class has a constructor with arguments, the class must also " +
+               "have a default, no-argument constructor. The OpenJPA implementation automatically generates the " +
+               "no-argument constructor, but the EclipseLink implementation does not.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                Preconditions.or(
+                        new UsesType<>("javax.persistence.Entity", true),
+                        new UsesType<>("javax.persistence.MappedSuperclass", true)),
```

Comment:
I didn't see any mentioned of the `@MappedSuperclass` annotation in the OpenJPA to EclipseLink JPA Migration guide. Did you find additional information that suggested that we should be treating `@MappedSuperclass` the same as `@Entity`?

Reply:
Our rule in WAMT says:
> This rule scans for entity classes with the @Entity or @MappedSuperclass annotations. Entity classes with constructors that do not also have a no-argument constructor are flagged. The quick fix adds an empty no-argument constructor to the class.

Should that be taken to mean that `@MappedSuperclass` still isn't modified?

Reply:
After we discussed, seems we should add the default constructor to a MappedSuperclass